### ISSUE TITLE
[Windows 7] NSIS.Library.RegTool.v3.[..].exe endlessly spinning component registration on boot

### ISSUE THREAD

**Issue by AnrDaemon** [owncloud/client#5058]
### Expected behaviour

No idea, what should have happened, but most certainly it shouldn't block OS startup for 15 minutes.
### Actual behaviour

NSIS.Library.RegTool.v3.{FD7D9A09-6009-41F8-AA50-A1076BCCCEB4}.exe was sitting there repeatedly spawning `regsvr32.exe /S OCContextMenu_x64.dll` and `regsvr32 /S exe OCOverlays_x64.dll`
### Steps to reproduce

No idea. I just rebooted the system, and this happened.
### Server configuration

Not applicable.
### Client configuration

Client version: 2.2.2 b6192

Operating system: Win7 Pro 64bit.

OS language: RU

Installation path of client: Default (%ProgramFilesx86%)
### Logs

No logs, client didn't started yet.

**Comment by guruz** [comment]
So this happened up to 15 minutes?
Or in an endless loop the same command?

**Comment by AnrDaemon** [comment]
I had to headshot the process to let user login continue. My patience doesn't run this short, when it is obvious the process is going in circles.

**Comment by guruz** [comment]
We need more info to help you with this.
Did you upgrade from a previous ownCloud version?
Is this a new install?
etc

**Comment by AnrDaemon** [comment]
I said, I have no idea. I, for all that I know, did not do anything special. I just rebooted the system after installing a few Windows updates, and logged into my desktop.
Then I watched my cursor derping on a nearly empty desktop, trying to decide between stateNormal and stateAppWait, and chose to investigate. When I opened a task manager, the NSIS Regtool stood out of the six or so processes running as a logged user. Between Shell/USBDLM/RMS/etc. apps started by the services, it was the only app started by the shell itself.
May be it upgraded by itself in a previous boot, and stuffed that trojan into RunOnce a few weeks ago. I should really look into disabling its autoupdate for good, no autoupdate ever helped, but those few causing trouble, I forever remember.
There was the trustedinstaller running along, but I waited for it to finish settling the updates, still didn't helped. I then shot the regtool, and checked if there's anything suspicious in registration of the both OCX. Both reported successful registration when launched bu hands.
Bottom line, I have no idea, why it ever been there. Installer itself is supposed to run these registrations, no?

**Comment by guruz** [comment]
Maybe something in the NSIS bugtracker? https://sourceforge.net/p/nsis/bugs/

FYI @danimo

**Comment by danimo** [comment]
@AnrDaemon Yes, it is supposed to do these registrations, regardless if its an auto or manual update. It needs to swap DLLs that Explorer loads, and there is no other clean way to ensure that the files can be swapped, because explorer must not be active at this point. Hence the RunOnce approach. This is done by all installers btw, and we're using a standard NSIS feature.

**Comment by guruz** [comment]
Moving to 2.3 unless new info comes from @AnrDaemon or others on how we can reproduce this to fix it or if this is a known NSIS issue or similar.

**Comment by guruz** [comment]
Please comment if this still happens with a recent oC server/client
